modules/core/src/main/scala/zio/kafka/streams/TopologySdk.scala

```diff
@@ -0,0 +1,69 @@
+package zio.kafka.streams
+
+import org.apache.kafka.common.serialization.Serde
+import org.apache.kafka.streams.Topology
+import org.apache.kafka.streams.scala.StreamsBuilder
+import org.apache.kafka.streams.scala.kstream.{ Consumed, KStream, Produced }
+import zio.prelude.State
+import zio.prelude.fx.ZPure
+
+trait TopologySdk {
+  def streamFromSource[K, V](
+    sourceSafeTopic: SafeTopic[K, V]
+  ): State[StreamsBuilder, KStream[K, V]] = {
```

Comment:
if you use only `State[StreamsBuilder, KStream[K, V]]`, how do you access the config or other envs inside the topology i.e. `RIO[KafkaStreamsConfig, Topology]`?

Reply:
I don't know, what about passing a `StoreBuilder` and a config (`StoreBuilder => StoreBuilder`) when you build the store? As normal parameters?

Reply:
that was just something off the top of my head, I'm pretty sure there are cases where you might want to access some values from the ZEnv